Doc/library/dbm.rst

```diff
@@ -346,61 +346,66 @@ to locate the appropriate header file to simplify building this module.
 
 --------------
 
-The :mod:`dbm.dumb` module provides a persistent dictionary-like interface which
-is written entirely in Python.  Unlike other modules such as :mod:`dbm.gnu` no
-external library is required.  As with other persistent mappings, the keys and
-values are always stored as bytes.
-
-The module defines the following:
+The :mod:`dbm.dumb` module provides a persistent :class:`dict`-like
+interface which is written entirely in Python.
+Unlike other :mod:`dbm` backends, such as :mod:`dbm.gnu` no
+external library is required.
+As with other :mod:`dbm` backends,
+the keys and values are always stored as :class:`bytes`.
 
+The :mod:`!dbm.dumb` module defines the following:
 
 .. exception:: error
 
    Raised on :mod:`dbm.dumb`-specific errors, such as I/O errors.  :exc:`KeyError` is
    raised for general mapping errors like specifying an incorrect key.
 
 
-.. function:: open(filename[, flag[, mode]])
+.. function:: open(filename, flag="c", mode=0o666)
 
-   Open a ``dumbdbm`` database and return a dumbdbm object.  The *filename* argument is
-   the basename of the database file (without any specific extensions).  When a
-   dumbdbm database is created, files with :file:`.dat` and :file:`.dir` extensions
-   are created.
+   Open a :mod:`!dbm.dumb` database.
+   The returned object behaves similar to a :term:`mapping`,
+   in addition to providing :meth:`~dumbdbm.sync` and :meth:`~dumbdbm.close`
+   methods.
 
-   The optional *flag* argument can be:
+   :param filename:
+      The basename of the database file (without extensions).
+      A new database creates the following files:
+      - :samp:`{filename}.dat`
+      - :samp:`{filename}.dir`
```

Comment:
Would not `:file:` role be more appropriate?

Reply:
IMO, no. I wanted to emphasise that the parameter in question is the *basename* of the generated files. I can do that better with :samp: than with :file:

Reply:
But `{filename}.dat` _is_ a file name.

Reply:
Here's the render. `:samp:` to the left, `:file:` to the right. It is not rendered as `{filename}.data`, Serhiy.

![Screenshot 2024-01-26 at 15 12 28](https://github.com/python/cpython/assets/13780613/7022e765-39a8-4ea2-84ac-58357c50a134)

Reply:
See the Sphinx docs for more information about `:samp:`:

- https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-samp

Reply:
According to the documentation, `filename` should be rendered as a variable part.
https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-file

Reply:
```suggestion
      - :file:`{filename}.dat`
      - :file:`{filename}.dir`
```